draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,528 +1,448 @@
+   added at the server and client end points.  The discussions should
+   then consider fairness with non-coding solutions.
+
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 4]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+3.3.  Objective of the document
+
+   Section 2 presents that FEC and CC can be seen as two separate
+   channels.  In practice, implementations may mix the signals that are
+   exchanged on these channels.  This document discusses how FEC and CC
+   relate.  [NK: We may want to later add that we provide
+   recommendations - but for the moment, we may just list the server
+   side coding solutions in the layering approach without providing
+   recommendations for each case]
+
+4.  FEC above the transport
+
+   Figure 3 illustrates the FEC above the transport case.
+
+                 |                  --------                  ^
+                 | data             |      | data,            | sending
+                 |                  |      | requirements     | rate (or
+                 v                  |      v                  | window)
+  +-----------------------------+   |    +--------------------------+
+  |             FEC             |        | Transport (including CC) |
+  +-----------------------------+   |    +--------------------------+
+    ^                   |           |              ^
+    | information       | coded     |              | network
+    | about repaired    | data      |              | measurements
+    | ratio             -------------              |
+
+                     Figure 3: FEC above the transport
+
+   The coded packets are sent within what the congestion window allows.
+
+   TODO : ADD MORE TEXT
+
+5.  FEC within the transport
+
+   Figure 4 illustrates the FEC within the transport case.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 5]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+                          |
+                          | data, requirements
+                          |
+                          v
+           +---------------------+
+           | Transport           |
+           | +-----+  +-----+    | coded data
+           | | FEC |  | CC  |    |
+           | +-----+  +-----+    | ----->
+           |                     |
+           +---------------------+
+                   ^          ^
+                   |          |
+           information    network
+           about          measurements
+           repaired ratio
+
+                      Figure 4: FEC in the transport
+
+   The coded packets are sent within what the congestion window allows,
+   such as in [CTCP].  Examples of the solution would be sending coded
+   packets when there is no more data to transmit or preferably send
+   coded packets instead of the following packets in the send buffer.
```

Comment:
the "sending coded packets when there is no more data to transmit" idea is GREAT!

Reply:
It is not mine ! :)

Reply:
Oh my god, I hope it wasn't from me, that would be self-praise  :(  I really really didn't want to do that!